pkg/relayer/session/session.go

```diff
@@ -311,19 +311,38 @@ func (rs *relayerSessionsManager) validateConfig() error {
 
 // waitForBlock blocks until the block at the given height (or greater) is
 // observed as having been committed.
-func (rs *relayerSessionsManager) waitForBlock(ctx context.Context, height int64) client.Block {
+func (rs *relayerSessionsManager) waitForBlock(ctx context.Context, targetHeight int64) client.Block {
 	// Create a cancellable child context for managing the CommittedBlocksSequence lifecycle.
-	// Since the subscription is no longer needed after the block it is looking for
+	// Since the committedBlocksObserver is no longer needed after the block it is looking for
 	// is reached, cancelling the child context at the end of the function will stop
 	// the subscriptions and close the publish channel associated with the
 	// CommittedBlocksSequence observable which is not exposing it.
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
-	subscription := rs.blockClient.CommittedBlocksSequence(ctx).Subscribe(ctx)
+	committedBlocksObs := rs.blockClient.CommittedBlocksSequence(ctx)
+	committedBlocksObserver := committedBlocksObs.Subscribe(ctx)
+
+	// minNumReplayBlocks is the number of blocks that MUST be in the block client's
+	// replay buffer such that the target block can be observed. Plus one is necessary
+	// for the "oldest" boundary to include targetHeight. If minNumReplayBlocks is negative,
+	// no replay is necessary and the replay buffer will be ignored.
```

Comment:
Lmk if you disagree with this style. It helps me separate and actually understand what's happening.
```suggestion
	// minNumReplayBlocks is the number of blocks that MUST be in the block client's
	// replay buffer such that the target block can be observed.
	//
	// Plus one is necessary for the "oldest" boundary to include targetHeight.
	//
	// If minNumReplayBlocks is negative, no replay is necessary and the replay buffer will be ignored.
```